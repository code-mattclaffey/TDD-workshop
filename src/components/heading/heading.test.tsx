import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Heading } from '.';

afterEach(() => {
    cleanup();
});

describe('<Heading />', () => {
    it('should match h1 snapshot', () => {
        const { container } = render(
            <Heading as="h1" type="h1">
                Headline 1
            </Heading>,
        );
        expect(container).toMatchSnapshot();
    });

    it('should match h2 snapshot', () => {
        const { container } = render(<Heading>Headline 2</Heading>);
        expect(container).toMatchSnapshot();
    });

    it('should match h3 snapshot', () => {
        const { container } = render(
            <Heading as="h3" type="h3">
                Headline 3
            </Heading>,
        );
        expect(container).toMatchSnapshot();
    });

    it('should match h4 snapshot', () => {
        const { container } = render(
            <Heading as="h4" type="h4">
                Headline 4
            </Heading>,
        );
        expect(container).toMatchSnapshot();
    });

    it('should match h5 snapshot', () => {
        const { container } = render(
            <Heading as="h5" type="h5">
                Headline 5
            </Heading>,
        );
        expect(container).toMatchSnapshot();
    });

    it('should match h6 snapshot', () => {
        const { container } = render(
            <Heading as="h6" type="h6">
                Headline 6
            </Heading>,
        );
        expect(container).toMatchSnapshot();
    });
});
