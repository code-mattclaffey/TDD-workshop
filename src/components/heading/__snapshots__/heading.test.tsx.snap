// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match h1 snapshot 1`] = `
<div>
  <h1
    class="e-heading e-heading--h1"
  >
    Headline 1
  </h1>
</div>
`;

exports[`<Heading /> should match h2 snapshot 1`] = `
<div>
  <h2
    class="e-heading e-heading--h2"
  >
    Headline 2
  </h2>
</div>
`;

exports[`<Heading /> should match h3 snapshot 1`] = `
<div>
  <h3
    class="e-heading e-heading--h3"
  >
    Headline 3
  </h3>
</div>
`;

exports[`<Heading /> should match h4 snapshot 1`] = `
<div>
  <h4
    class="e-heading e-heading--h4"
  >
    Headline 4
  </h4>
</div>
`;

exports[`<Heading /> should match h5 snapshot 1`] = `
<div>
  <h5
    class="e-heading e-heading--h5"
  >
    Headline 5
  </h5>
</div>
`;

exports[`<Heading /> should match h6 snapshot 1`] = `
<div>
  <h6
    class="e-heading e-heading--h6"
  >
    Headline 6
  </h6>
</div>
`;
